### PR TITLE
Support Fuse Mount with Worker to Reduce RPCs

### PR DESCRIPTION
**Pull request by apc999**
Fix https://github.com/Alluxio/alluxio/issues/12830

Co-authored-by: Lu Qiu <luqiujob@gmail.com>

### CODE REVIEW

File: core/client/fs/src/main/java/alluxio/client/block/stream/BlockWorkerDataWriter.java [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+    if (mReservedBytes < pos() + buf.readableBytes()) {
+      try {
+        long bytesToReserve = Math.max(mBufferSize, pos() + buf.readableBytes() - mReservedBytes);
+        // Allocate enough space in the existing temporary block for the write.
+        mBlockWorker.requestSpace(mSessionId, mBlockId, bytesToReserve);
+      } catch (Exception e) {
+        throw new IOException(e);
+      }
+    }
+    long append = mBlockWriter.append(buf);
```
Comment (LuQQiu) [inline core/client/fs/src/main/java/alluxio/client/block/stream/BlockWorkerDataWriter.java:105]: If `writeChunk` throws an exception, will the DataWriter guarantee to be cancelled?

**Review by LuQQiu — APPROVED** [review]
LGTM! Thanks @apc999 for the amazing work!

**Comment by apc999** [comment]
alluxio-bot, merge this please

**Comment by apc999** [comment]
alluxio-bot, feature-merge this please

**Pull request merged**